### ISSUE TITLE
Grid: Bad aspect of checkbox column when grid is populated after the page loading

### ISSUE THREAD

**Issue by pinonpierre** [vaadin/framework#9509]
In Vaadin 8.0.6 (Same issue on 8.1.0.beta1 too), I have a bad aspect of checkbox column when grid is populated after the page loading

Code to reproduce the problem:

```
        Grid<String> grid = new Grid<>();
        grid.setSelectionMode(Grid.SelectionMode.MULTI);
        grid.addColumn(string -> string);

        mainLayout.addComponent(grid);

        Button button = new Button("Do it!");
        button.addClickListener(event1 -> grid.setItems("blue", "red", "green", "yellow", "purple"));
        solutionButtons.addComponent(button);
```
![bad](https://user-images.githubusercontent.com/4507910/26931853-1fa5a8c8-4c62-11e7-8798-2ea0f65f097e.PNG)

No problem with this code:
```
        Grid<String> grid = new Grid<>();
        grid.setSelectionMode(Grid.SelectionMode.MULTI);
        grid.addColumn(string -> string);
        grid.setItems("blue", "red", "green", "yellow", "purple")

        mainLayout.addComponent(grid);
```
![good](https://user-images.githubusercontent.com/4507910/26931855-22e0469c-4c62-11e7-8f32-0290a9ab6975.PNG)

**Comment by TatuLund** [comment]
Is this related to this one https://github.com/vaadin/framework/issues/10239

**Comment by tsuoanttila** [comment]
Does this work with the most recent `8.3-SNAPSHOT`?

**Comment by tsuoanttila** [comment]
Tested and seems to be fixed along with #10239. Marking as duplicate.
